### PR TITLE
[FIX] payment_custom: avoid empty pending message

### PR DESCRIPTION
**Pull request by hote-odoo**
[FIX] payment_custom: avoid empty pending message
If `payment_custom` is installed before `account` not account information
is set up and the `pending_msg` remains as `False`. This implies that no
pending message is shown in the portal after transaction confirmation.
After this commit if `payment_custom` is installed before the `account`
module then the pending message will fall back to the default message
defined in the `payment` module.

Task - 3000442

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_custom/models/payment_provider.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def _compute_view_configuration_fields(self):
 
     def _transfer_ensure_pending_msg_is_set(self):
         transfer_providers_without_msg = self.filtered(
```
Comment (AntoineVDV) [inline addons/payment_custom/models/payment_provider.py:42]: This variable name does not mean anything anymore.

File: addons/payment_custom/models/payment_provider.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def _compute_view_configuration_fields(self):
...
-            lambda p: p.code == 'custom' and not p.pending_msg)
+            lambda p: p.code == 'custom')
```
Comment (AntoineVDV) [inline addons/payment_custom/models/payment_provider.py:43]: The filter could now be done on one line.

File: addons/payment_custom/models/payment_provider.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def _compute_view_configuration_fields(self):
...
         if not transfer_providers_without_msg:
```
Comment (AntoineVDV) [inline addons/payment_custom/models/payment_provider.py:44]: Just merge this check within the `for` loop below.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Commit message:
- Is the problem that `payment_custom` is auto-installed?
- A [FIX] commit should clearly explain the bug and how it's being fixed.

File: addons/payment_custom/models/payment_provider.py [hunk 4]
```diff
@@ -39,16 +39,14 @@ def _compute_view_configuration_fields(self):
...
-            return
-
         account_payment = self.env['ir.module.module']._get('account_payment')
         if account_payment.state != 'installed':
             return
 
-        for provider in transfer_providers_without_msg:
+        for provider in self:
+            if provider.code != 'custom':
+                continue
```
Comment (AntoineVDV) [inline addons/payment_custom/models/payment_provider.py:48]:
```suggestion
        for provider in self.filtered(lambda p: p.code == 'custom'):
```

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: addons/payment_custom/models/payment_provider.py [hunk 5]
```diff
@@ -45,19 +45,24 @@ def _transfer_ensure_pending_msg_is_set(self):
...
-                f'<h4>{_("Communication")}</h4>' \
-                f'<p>{_("Please use the order name as communication reference.")}</p>' \
-                f'</div>'
+            if account_payment.state != 'installed':
+                provider.pending_msg = f'<div>' \
+                    f'<h3>{_("Please use the following transfer details")}</h3>' \
+                    f'<h4>{_("Bank Account")}</h4>' \
+                    f'<h4>{_("Communication")}</h4>' \
+                    f'<p>{_("Please use the order name as communication reference.")}</p>' \
+                    f'</div>'
```
Comment (Feyensv) [inline addons/payment_custom/models/payment_provider.py:55]: Could be assigned in batch to all the providers, since the values will be the same.

Comment (hote-odoo) [reply addons/payment_custom/models/payment_provider.py]: thanks :)

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**